cdk/lib/support-reminders.ts

```diff
@@ -1,15 +1,247 @@
-import { join } from "path";
-import type { GuStackProps } from "@guardian/cdk/lib/constructs/core";
-import { GuStack } from "@guardian/cdk/lib/constructs/core";
-import type { App } from "aws-cdk-lib";
-import { CfnInclude } from "aws-cdk-lib/cloudformation-include";
+import path from "path";
+import {GuApiGatewayWithLambdaByPath, GuScheduledLambda} from "@guardian/cdk";
+import type {GuStackProps} from "@guardian/cdk/lib/constructs/core";
+import {GuStack, GuStringParameter} from "@guardian/cdk/lib/constructs/core";
+import {GuVpc} from "@guardian/cdk/lib/constructs/ec2";
+import {GuLambdaFunction} from "@guardian/cdk/lib/constructs/lambda";
+import type {App} from "aws-cdk-lib";
+import {CfnBasePathMapping, CfnDomainName, Cors} from "aws-cdk-lib/aws-apigateway";
+import {SecurityGroup} from "aws-cdk-lib/aws-ec2";
+import {Schedule} from "aws-cdk-lib/aws-events";
+import {Effect, ManagedPolicy, Policy, PolicyStatement} from "aws-cdk-lib/aws-iam";
+import {Runtime} from "aws-cdk-lib/aws-lambda";
+import {CfnRecordSet} from "aws-cdk-lib/aws-route53";
+import {CfnInclude} from "aws-cdk-lib/cloudformation-include";
+
+export interface SupportRemindersProps extends GuStackProps {
+	certificateId: string;
+	domainName: string;
+	hostedZoneId: string;
+	datalakeBucket: string;
+}
 
 export class SupportReminders extends GuStack {
-  constructor(scope: App, id: string, props: GuStackProps) {
-    super(scope, id, props);
-    const yamlTemplateFilePath = join(__dirname, "../..", "cfn.yaml");
-    new CfnInclude(this, "YamlTemplate", {
-      templateFile: yamlTemplateFilePath,
-    });
-  }
+	constructor(scope: App, id: string, props: SupportRemindersProps) {
+		super(scope, id, props);
+
+
+		// ---- CFN template resources ---- //
+		const yamlTemplateFilePath = path.join(__dirname, "../..", "cfn.yaml");
+		new CfnInclude(this, "YamlTemplate", {
+			templateFile: yamlTemplateFilePath,
+		});
+
+
+		// ---- Parameters ---- //
+		const securityGroupToAccessPostgres = new GuStringParameter(
```

Comment:
As discussed in person, we could consider passing this in (as a string) via `SupportRemindersProps` in a future PR. 

Do you think it's OK to include a security group id here @akash1810 @nicl?

Reply:
I'm going to go ahead and merge this PR, but it would be good to get rid of this parameter in a follow-up!